src/main/java/com/squareup/javapoet/ClassName.java

```diff
@@ -122,7 +122,7 @@ public static ClassName bestGuess(String classNameString) {
 
     // Add the package name, like "java.util.concurrent", or "" for no package.
     int p = 0;
-    while (p < classNameString.length() && Util.isLowerCase(classNameString.charAt(p))) {
+    while (p < classNameString.length() && Character.isLowerCase(classNameString.charAt(p))) {
```

Comment:
As an experiment, in NetBeans I wrote this in a method:
    int \ud835\udc00 = 5;
    int 𝐀 = 5;
It correctly complained that the second 𝐀 was a redefinition of the first (hidden in a surrogate pair). :-)